### PR TITLE
Moment of Inertia Datadefs.hs

### PR DESCRIPTION
**Pull request by oluowoj**
@JacquesCarette please ignore PR #1425 and use this.

I added MomemtOfInertia as a new data definition in the gamePhy example

### CODE REVIEW

**Comment by oluowoj** [comment]
related to PR #1425(obsolete)

File: code/stable/gamephys/SRS/Chipmunk_SRS.tex [hunk 1]
```diff
@@ -892,6 +894,38 @@ \subsubsection{Data Definitions}
...
+Units & kg$\text{m}^{2}$
+\\ \midrule \\
+Equation & \begin{displaymath}
+           \mathbf{I}=\displaystyle\sum{{m_{j}} {r_{j}}^{2}}
+           \end{displaymath}
+\\ \midrule \\
+Description & \begin{symbDescription}
+              \item{$\mathbf{I}$ is the moment of inertia (kg$\text{m}^{2}$)}
+              \item{${m_{j}}$ is the mass of the j-th particle (kg)}
+              \item{${r_{j}}$ is the distance between the j-th particle and the angle of rotation (m)}
```
Comment (smiths) [inline code/stable/gamephys/SRS/Chipmunk_SRS.tex]: This is the distance between the jth particle and the axis of rotation.

File: code/drasil-example/Drasil/GamePhysics/Unitals.hs [hunk 2]
```diff
@@ -203,7 +203,7 @@ normalLen = ucs' (dccWDS "length of the normal vector" (compoundPhrase'
                   (Concat [Atomic "||",(eqSymb QM.normalVect), Atomic "||"]) Real metre
 
 rRot = ucs' (dccWDS "r_j" (compoundPhrase' (QP.distance ^. term)
-                (cn "between the j-th particle and the angle of rotation")) (phrase QP.distance)) 
+                (cn "between the j-th particle and the axis of rotation")) (phrase QP.distance))
```
Comment (smiths) [inline code/drasil-example/Drasil/GamePhysics/Unitals.hs]: Looks good

**Comment by JacquesCarette** [comment]
and of course, a master merge is needed. But otherwise this does look ready.

**Comment by JacquesCarette** [comment]
So is this meant to be a 'live' branch that should be merged in? If so, then the fact that it doesn't build should be fixed. And why has this been in this state for 5 days?

**Comment by oluowoj** [comment]
merge completed

**Comment by JacquesCarette** [comment]
But since other things got merged in, yet another merge is needed!

**Comment by JacquesCarette** [comment]
Fails hLint.  Did you build locally before committing? You would have spotted those.

File: code/drasil-example/Drasil/GamePhysics/DataDefs.hs [hunk 3]
```diff
@@ -367,11 +367,11 @@ momentOfInertia :: QDefinition
 momentOfInertia = mkQuantDef QP.momentOfInertia momentOfInertiaEqn
 
 momentOfInertiaEqn :: Expr
-momentOfInertiaEqn = (sumAll (Atomic "i")((sy massI)*(sy  rRot $^ 2)))
+momentOfInertiaEqn = sumAll (Atomic "i") (sy massI) * (sy rRot $^ 2)
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: This change is incorrect, it changes the meaning. This is clearer in the latex/html where there's a variable j outside of its binder.

File: code/stable/gamephys/SRS/Chipmunk_SRS.tex [hunk 4]
```diff
@@ -1002,7 +1002,7 @@ \subsubsection{Data Definitions}
 Units & kg$\text{m}^{2}$
 \\ \midrule \\
 Equation & \begin{displaymath}
-           \mathbf{I}=\displaystyle\sum{{m_{j}} {r_{j}}^{2}}
+           \mathbf{I}=\left(\displaystyle\sum{{m_{j}}}\right) {r_{j}}^{2}
```
Comment (JacquesCarette) [inline code/stable/gamephys/SRS/Chipmunk_SRS.tex]: The sum should scope over both.

File: code/drasil-example/Drasil/GamePhysics/DataDefs.hs (hunk 3 above)
Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Oh yeah, I think I pushed this change in error, I was just playing around with the Hlint suggestions and I think I hit 'Enter' on scrolling to git push

File: code/stable/gamephys/SRS/Chipmunk_SRS.tex (hunk 4 above)
Comment (oluowoj) [reply code/stable/gamephys/SRS/Chipmunk_SRS.tex]: Yes you are right

File: code/drasil-example/Drasil/GamePhysics/DataDefs.hs [hunk 5]
```diff
@@ -351,6 +356,22 @@ kEnergyDesc :: Sentence
...
+
+momentOfInertiaDD :: DataDefinition
+momentOfInertiaDD = ddNoRefs momentOfInertia [{-- Derivation --}] "momentOfInertia"
+ [momentOfInertiaDesc, makeRef2S assumpOT] 
+
+momentOfInertia :: QDefinition
+momentOfInertia = mkQuantDef QP.momentOfInertia momentOfInertiaEqn
+
+momentOfInertiaEqn :: Expr
+momentOfInertiaEqn = (sumAll (Atomic "i") (sy massI) * (sy rRot $^ 2))
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Those outer () are still causing `hLint` to be unhappy.

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: HLint suggestion as you said changes the meaning, but I took out the outer brackets.
image

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: @JacquesCarette , if I don't use the HLint suggestion for any reason, is the build always going to fail?

Comment (JacquesCarette) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Probably.  I think the correct expression here should be
```haskell
sumAll (Atomic "i") $ sy massI * (sy rRot $^2)
```

Comment (JacquesCarette) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: (but I have not tested it)

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Okay so why did I not try this earlier.... I thought about it but did not try it out thinking  the "()" is as good as "$". So HLint does not like brackets.

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: I will push and see if this works

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Yay, it worked. Thanks

Comment (JacquesCarette) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Note that you should be able to test this without pushing - I do believe that 'make' runs hlint -- right @Mornix ?

Comment (oluowoj) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: Usually when my local build is successful I still get Hlint errors if the build fails.

Comment (JacquesCarette) [reply code/drasil-example/Drasil/GamePhysics/DataDefs.hs]: We should definitely make sure that hlint runs locally too.

**Pull request merged**